clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -781,6 +782,24 @@ raw2trace_t::do_conversion()
     return "";
 }
 
+bool
+raw2trace_t::record_encoding_emitted(void *tls, app_pc pc)
+{
+    auto tdata = reinterpret_cast<raw2trace_thread_data_t *>(tls);
+    if (tdata->encoding_emitted.find(pc) != tdata->encoding_emitted.end())
+        return false;
+    tdata->encoding_emitted.insert(pc);
+    tdata->last_encoding_emitted = pc;
+    return true;
+}
+
+void
+raw2trace_t::rollback_last_encoding(void *tls)
+{
+    auto tdata = reinterpret_cast<raw2trace_thread_data_t *>(tls);
+    tdata->encoding_emitted.erase(tdata->last_encoding_emitted);
```

Comment:
I added comments in the doc, header, and impl that it can only be called once between calls to record_encoding_emitted().  I also added a clear to nullptr for last_encoding_emitted after the rollback and an assert on entry that it's not nullptr to catch a 2nd call.